src/App.tsx

```diff
@@ -1,26 +1,172 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import { useState, useEffect, FC } from 'react';
+import cn from 'classnames';
+
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import Header from './components/Header';
+import { Footer } from './components/Footer';
+import TodoList from './components/TodoList';
+import { Status } from './types/Status';
+import { getTodos, postTodo, deleteTodoById } from './api/todos';
+import { Todo } from './types/Todo';
+import { handleError } from './utils/HandleErrorShow';
+
+export const App: FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [status, setStatus] = useState<Status>(Status.All);
+  const [errorMessage, setErrorMessage] = useState<string>('');
+  const [fieldQuery, setFieldQuery] = useState<string>('');
+  const [isLoading, setIsLoading] = useState<boolean>(false);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isFocusInput, setFocusInput] = useState<boolean>(false);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        handleError('Unable to load todos', setErrorMessage);
+      });
+  }, []);
 
-const USER_ID = 0;
+  useEffect(() => {});
 
-export const App: React.FC = () => {
   if (!USER_ID) {
     return <UserWarning />;
   }
 
+  const itemsLeft = todos.filter(({ completed }) => !completed).length;
+  const haveCompletedTodos = todos.some(({ completed }) => completed);
+  const allCompleted = todos.every(({ completed }) => completed);
+  const filteredTodos = todos.filter(task => {
+    switch (status) {
+      case Status.Active:
+        return !task.completed;
+      case Status.Completed:
+        return task.completed;
+      case Status.All:
+      default:
+        return true;
+    }
+  });
+
+  const addTodo = (e: React.FormEvent) => {
+    e.preventDefault();
+
+    setIsLoading(true);
+    const trimTodo = fieldQuery.trim();
+
+    if (!trimTodo) {
+      handleError('Title should not be empty', setErrorMessage);
+      setFocusInput(true);
+      setIsLoading(false);
+
+      return;
+    }
+
+    const newTodo: Todo = {
+      id: 0,
+      userId: USER_ID,
+      title: trimTodo,
+      completed: false,
+      status: status,
+    };
+
+    const copyNewTodo = { ...newTodo, id: todos.length + 1 };
+
+    setTempTodo(newTodo);
+
+    postTodo(newTodo)
+      .then(() => {
+        setTodos([...todos, copyNewTodo]);
```

Comment:
If you need to update state depending on the previous one use functional `setState`